### PR TITLE
Adapt calculation of scale to work independently of the underlying HTML slide system

### PR DESCRIPTION
**Pull request by dontcallmedom**

### CODE REVIEW

File: i-slide.js [hunk 1]
```diff
@@ -248,41 +248,51 @@ class ISlide extends HTMLElement {
...
       htmlEl.appendChild(bodyEl);
+      const styleLoaded = Promise.all([...headEl.querySelectorAll("link[rel~=stylesheet]")].map(l => new Promise((res) => {
+        l.addEventListener("load", res);
+        l.addEventListener("error", res);
+      })));
+
       this.shadowRoot.append(htmlEl);
+      styleLoaded.then(() => {
+        // We need the slide to be rendered to measure its pixel dimensions
+        // through window.getComputedStyle
```
Comment (dontcallmedom) [inline i-slide.js:279]:
```suggestion
```

File: i-slide.js [hunk 2]
```diff
@@ -248,41 +248,50 @@ class ISlide extends HTMLElement {
...
       const htmlEl = document.createElement('html');
       htmlEl.appendChild(headEl);
       htmlEl.appendChild(bodyEl);
+      const styleLoaded = Promise.all([...headEl.querySelectorAll("link[rel~=stylesheet]")].map(l => new Promise((res) => {
+        l.addEventListener("load", res);
+        l.addEventListener("error", res);
+      })));
+
       this.shadowRoot.append(htmlEl);
+      styleLoaded.then(() => {
```
Comment (tidoust) [inline i-slide.js]: Editorial: why not `await styleLoaded` to avoid introducing a new level of indentation?

File: i-slide.js [hunk 3]
```diff
@@ -248,41 +248,54 @@ class ISlide extends HTMLElement {
...
+        l.addEventListener("error", resolve);
+        styleLoadedPromises.push(p);
+      });
       // Attach HTML document to shadow root
       const htmlEl = document.createElement('html');
       htmlEl.appendChild(headEl);
       htmlEl.appendChild(bodyEl);
+
       this.shadowRoot.append(htmlEl);
+      return Promise.all(styleLoadedPromises).then(() => {
```
Comment (tidoust) [inline i-slide.js:281]: Editorial: why not `await Promise.all(styleLoadedPromises)` to avoid introducing a new level of indentation?

**Review by tidoust — APPROVED** [review]
Looks good to me with one editorial comment.

File: i-slide.js (hunk 3 above)
Comment (dontcallmedom) [reply i-slide.js]: done

**Pull request merged**